fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/BaseRepeatedField.java

```diff
@@ -82,15 +82,24 @@ public int hashCode() {
         return super.hashCode() + emptyMode.hashCode();
     }
 
+    /**
+     * Base implementation of {@link #planHash}.
+     * This implementation makes each concrete subclass implement its own version of {@link #planHash} so that they are
+     * guided to add their own class modifier (See {@link com.apple.foundationdb.record.ObjectPlanHash ObjectPlanHash}).
+     * This implementation is meant to give subclasses common functionality for their own implementation.
+     * @param hashKind the plan hash kind to use
+     * @param baseHash the subclass' base hash (concrete identifier)
+     * @param hashables the rest of the subclass' hashable parameters (if any)
+     * @return the plan hash value calculated
+     */
     @Override
-    public int planHash(@Nonnull final PlanHashKind hashKind) {
+    protected int basePlanHash(@Nonnull final PlanHashKind hashKind, ObjectPlanHash baseHash, Object... hashables) {
         switch (hashKind) {
             case LEGACY:
-                return super.planHash(hashKind) + emptyMode.ordinal();
+                return super.basePlanHash(hashKind, baseHash) + emptyMode.ordinal();
             case FOR_CONTINUATION:
             case STRUCTURAL_WITHOUT_LITERALS:
-                // No BASE_HASH since this is abstract
-                return PlanHashable.objectsPlanHash(hashKind, super.planHash(hashKind), emptyMode.ordinal());
+                return super.basePlanHash(hashKind, baseHash, emptyMode.ordinal(), hashables);
```

Comment:
Since there is no question of compatibility, would it be better to treat this enum like others and so using the name instead of using the ordinal?

Reply:
Done.